### DISCUSSION TITLE
How do you multiple values in function

### DISCUSSION THREAD

**Discussion by brentrobbins** [supabase/supabase#7873] [Questions] [answered]
I am able to update with a single value, but I can't for the life of my sort out how to update multiple values in a function. Here's the code I have working for a single value:
```
begin
  update auth.users
  set raw_user_meta_data = jsonb_set ("raw_user_meta_data"::jsonb, '{first_name}', to_jsonb(new.first_name::text))
  where id = new.id;

  return new;
end;
```
 
How would you add a `last_name` value to be updated/added as well? I really appreciate the help!

**Comment by GaryAustin1** [comment]
Edit:  Deleting this as you are not setting multiple columns.

**Comment by GaryAustin1** [comment, marked as answer]
Something like `set raw_user_meta_data = raw_user_meta_data || jsonb_build_object('first_name' ,new.first_name, 'last_name', new.last_name)` should work.

It should replace the values for first or last name if they exist, otherwise add them.

**Reply by brentrobbins** [reply]
Sweet! That worked like a charm. Thank you @GaryAustin1

**Reply by carson2222** [reply]
Yep thanks, 'jsonb_build_object' function is a perfect go-for in this case. :)
